src/app.js

```diff
@@ -0,0 +1,100 @@
+const city = document.querySelector('.city');
+const country = document.querySelector('.country');
+const tempBtn = document.querySelector('.temperature');
+const convertToCelsius = (tempInKelvin) => Math.floor(tempInKelvin + -273.15);
+const convertToFahrenheit = (tempInKelvin) => Math.floor(tempInKelvin + -459.67);
+
+const renderData = (data) => {
+  
+  if (tempBtn.classList.contains('fahrenheit')) {
+    tempNum.innerText = convertToFahrenheit(data.main.temp);
+    document.querySelector('.temp-degree').innerText = '';
+    document.querySelector('.temp-celsius').innerText = 'F';
+  } else {
+    tempNum.innerText = convertToCelsius(data.main.temp);
+    document.querySelector('.temp-degree').innerText = 'o';
+    document.querySelector('.temp-celsius').innerText = 'C';
+  }
+
+  tempDescription.innerText = data.weather[0].description;
+  city.innerText = data.name;
+  country.innerText = data.sys.country;
+  noLocationFound.classList.remove('no-location');
+  noLocationFound.classList.add('found-location');
+};
+
+const getWeatherAtLocation = (inputLocation) => {
+  const weatherApi = `http://api.openweathermap.org/data/2.5/weather?q=${inputLocation}&APPID=2874e0623c8807994e18916c8cd78f21`;
+  fetch(weatherApi, { mode: 'cors' })
+    .then((response) => response.json())
+    .then((data) => {
+      if (data.main == undefined) {
+        noLocationFound.innerText = 'Location weather unavailable';
+        noLocationFound.classList.remove('found-location');
+        noLocationFound.classList.add('no-location');
+      } else {
+        noLocationFound.innerText = `Weather in ${data.name}`;
+        renderData(data);
+      }
+    });
+};
+
+const listenToTempChange = () =>{
+  tempBtn.addEventListener('click', () => {
+    tempBtn.classList.toggle('fahrenheit');
+    const cityInput = document.querySelector('#get-location-id').value;
+    getWeatherAtLocation(cityInput);
+  });
+}
+
+
+
+const showWeatherAtUserLat = (lat, lng) => {
+  const weatherApi = `http://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${lng}&APPID=2874e0623c8807994e18916c8cd78f21`;
+  fetch(weatherApi, { mode: 'cors' })
+    .then((response) => response.json())
+    .then((data) => {
+      noLocationFound.innerText = `You're currently in ${data.name}`;
+      noLocationFound.classList.add('found-location');
+      renderData(data);
+    });
+};
+
+
+const runForm = () => {
+  const submitForm = document.querySelector('.get-location');
+  submitForm.addEventListener('submit', (event) => {
+    event.preventDefault();
+    const cityInput = document.querySelector('#get-location-id').value;
+    getWeatherAtLocation(cityInput);
+  });
+};
+
+const grabUserLocation = (lat, lng) => {
+  if (lat == undefined && lng == undefined) {
+    noLocationFound.innerText = 'Click \'Allow\' to enable location';
+  } else {
+    showWeatherAtUserLat(lat, lng);
+  }
+};
+
+const getCurrentLocation = () => {
+  // Get the latitude and the longitude;
```

Comment:
**Kindly remove all commented code, you can always use git history to access previous code right :slightly_smiling_face:**

Comments: It’s better to delete commented code. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))